analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/TokenTypeAnalyzerTest.Classifier.cs

```diff
@@ -1286,4 +1286,26 @@ public void M(object o)
                 }
             }
             """, allowSemanticModel);
+
+    [DataTestMethod]
+    [DataRow("var d = [u:dateTimePointer]->Date;", false)]
+    [DataRow("var d = dateTimePointer->[u:Date];", false)]
```

Comment:
`UnknownTokentype` here means the classification of the token isn't any of the other TokenTypes. It doesn't have anything to do with the actual .Net types involved. The enum is badly named. It should be called TokenClassification instead of TokenType.

Also:
For all identifier tokens, there can only be two possible TokenType outcomes: "TypeName" or "UnknownTokentype" *). If we can conclude that the token can never bind to a type, we are done. A pointer can never be a type. (On a normal member access, we can not be sure: `Math.Pi` here `Math` can bind to a type or a property. Here we need to consult the semantic model for `Math` and see what it binds to.

*) One exception is "value" in property setters, which is classified as a keyword.